validator/block_validator.go

```diff
@@ -500,6 +514,23 @@ func (v *BlockValidator) sendValidations(ctx context.Context) {
 			v.ProcessBatches(v.globalPosNextSend.BatchNumber, [][]byte{seqMsg})
 			seqBatchEntry = seqMsg
 		}
+		v.blockMutex.Lock()
+		if v.lastBlockValidatedUnknown {
+			firstMsgInBatch := arbutil.MessageIndex(0)
+			if v.globalPosNextSend.BatchNumber > 0 {
+				var err error
+				firstMsgInBatch, err = v.inboxTracker.GetBatchMessageCount(v.globalPosNextSend.BatchNumber - 1)
+				if err != nil {
+					log.Error("validator couldnt read message count", "err", err)
+					return
+				}
+			}
+			v.lastBlockValidated = uint64(arbutil.MessageCountToBlockNumber(firstMsgInBatch+arbutil.MessageIndex(v.globalPosNextSend.PosInBatch), v.genesisBlockNum))
```

Comment:
This works and is tested in rinkeby - but I'm not sure why.

I expected this computation to be right for nextBlockToValidate - as globalPosition in the assertion should point to the next block - shouldn't it?

Reply:
I was stuck on this for a while too, but you're calling `MessageCountToBlockNumber`, which essentially corrects for this. The number you're passing in is the _message index_ of the next block, but the _message count_ of the last block's.